transaction/extra/src/signed_contingent_input.rs

```diff
@@ -142,17 +149,111 @@ impl SignedContingentInput {
                 if amount.value == 0 {
                     return Err(SignedContingentInputError::ZeroPartialFillChange);
                 }
+
+                result.partial_fill_change = Some(amount);
+
                 // Check that each output can actually be revealed
                 for partial_fill_output in rules.partial_fill_outputs.iter() {
-                    partial_fill_output.reveal_amount()?;
+                    let (amount, _) = partial_fill_output.reveal_amount()?;
+                    if amount.value == 0 {
+                        return Err(SignedContingentInputError::ZeroPartialFillOutput);
+                    }
+                    result.partial_fill_outputs.push(amount);
                 }
             } else if !rules.partial_fill_outputs.is_empty() || rules.min_partial_fill_value != 0 {
                 return Err(SignedContingentInputError::MissingPartialFillChange);
             }
         }
 
+        Ok(result)
+    }
+}
+
+/// This summary object is constructed during validation of an SCI, by recording
+/// all the Amount objects that we successfully unmask.
+#[derive(Clone, Debug, Default, PartialEq, Eq)]
+pub struct SignedContingentInputAmounts {
+    /// The amount of the pseudo-output, i.e. the true input signed over in this
+    /// SCI
+    pub pseudo_output: Amount,
+    /// The amounts of the required outputs
+    pub required_outputs: Vec<Amount>,
+    /// The amounts of the partial fill outputs.
+    pub partial_fill_outputs: Vec<Amount>,
+    /// The amount of hte partial fill change if present.
+    pub partial_fill_change: Option<Amount>,
+}
+
+impl SignedContingentInputAmounts {
+    /// Computes the hypothetical change in balances that will occur if we fill
+    /// this SCI to a certain degree
+    ///
+    /// Add the outputs and inputs to a BTreemap which functions as a balance
+    /// sheet. Outputs from the SCI are positive, and the value of the input
+    /// is negative.
+    ///
+    /// Arguments:
+    /// partial_fill_value: The amount of the partial_fill_change we want to
+    /// keep. This should be zero if this is not a partial fill SCI.
```

Comment:
:wrench: The use of "change" and "we" is a bit confusing.
I'm thinking this is saying that the `partial_fill_value` is the amount being taken/claimed by the counterparty?
```suggestion
    /// * `partial_fill_value` - The amount being taken by the counter party. 
    ///   This should be zero if this is not a partial fill SCI.
```

It feels like this argument should be an Option<> like the partial change on the SCI.

Reply:
in this documentation, it's better to use terms like "us" than "the counter party" because it's more clear to the caller, IMO. I will try to improve this though